src/qililab/qprogram/quantum_machines_compiler.py

```diff
@@ -450,7 +451,13 @@ def _handle_wait(self, element: Wait):
             if isinstance(element.duration, Variable)
             else max(element.duration, self.MINIMUM_TIME)
         )
-        qua.wait(duration, bus)
+
+        qua.wait(
+            duration / int(self.WAIT_COEFF)
```

Comment:
If `duration` is of type `Variable`, then the division `duration / int(self.WAIT_COEFF)` will not make sense. What needs to be done is change the values of the loop instead, inside the `_handle_for_loop`, `_handle_loop` and `_handle_parallel_loop` methods.

For example, the phases in the `_handle_for_loop` change as follows:
```
if element.variable.domain is Domain.Phase:
            start, stop, step = start / self.PHASE_COEFF, stop / self.PHASE_COEFF, step / self.PHASE_COEFF
```

Reply:
Nevermind, I just saw that `duration` is a qua variable, so this works of course. Well done